Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Receivers/InteractionReceiver.cs

```diff
@@ -0,0 +1,343 @@
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Input;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.SDK.Input.Handlers;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Receivers
+{
+    /// <summary>
+    /// An interaction receiver is simply a component that attached to a list of interactable objects and does something
+    /// based on events from those interactable objects. This is the base abstract class to extend from.
+    /// </summary>
+    public abstract class InteractionReceiver : BaseInputHandler,
+        IMixedRealityFocusChangedHandler,
+        IMixedRealityGestureHandler,
+        IMixedRealityInputHandler,
+        IMixedRealityGestureHandler<Vector2>,
+        IMixedRealityGestureHandler<Vector3>,
+        IMixedRealityGestureHandler<Quaternion>
+
+    {
+        #region Public Members
+        /// <summary>
+        /// List of linked interactable objects to receive events for
+        /// </summary>
+        [SerializeField]
+        [Tooltip( "Target interactable Object to receive events for" )]
+        private List<GameObject> interactables = new List<GameObject>();
+
+        public List<GameObject> Interactables
+        {
+            get { return interactables; }
+            private set { value = interactables; }
+        }
+
+        /// <summary>
+        /// Is Focus required to receive input events on this GameObject?
+        /// InteractionReceivers handle events for objects indirectly, in order for this class work properly, keep this set to false.
+        /// </summary>
+        public override bool IsFocusRequired
+        {
+            get { return base.IsFocusRequired; }
+            protected set { base.IsFocusRequired = value; }
+        }
+
+        /// <summary>
+        /// List of linked targets that the receiver affects
+        /// </summary>
+        [Tooltip( "Targets for the receiver to affect" )]
+        private List<GameObject> targets = new List<GameObject>();
+
+        public List<GameObject> Targets
+        {
+            get { return targets; }
+            private set { value = targets; }
+        }
+
+        #endregion
+
+        #region private variables
+        /// <summary>
+        /// When true, this interaction receiver will draw connections in the editor to Interactables and Targets
+        /// </summary>
+        [Tooltip( "When true, this interaction receiver will draw connections in the editor to Interactables and Targets" )]
+        [SerializeField]
+        private bool drawEditorConnections = true;
+        
+        #endregion
+
+        /// <summary>
+        /// On enable, set the BaseInputHandler's IsFocusRequired to false to receive all events.
+        /// </summary>
+        protected override void OnEnable()
```

Comment:
I'd much rather sort the regions by monobehaviors and interface implementations

Reply:
Fair enough - just trying to be consistent. Do we have region recommendations in the code guidance?

Reply:
Hmm, no but maybe it's something we should add.

Reply:
fixed.

Reply:
I'd take a look at any of the pointers to see how we've broken out the regions by interface